src/structures/Presence.js

```diff
@@ -292,7 +292,8 @@ class Activity {
         this.type === activity.type &&
         this.url === activity.url &&
         this.state === activity.state &&
-        this.details === activity.details)
+        this.details === activity.details &&
+        this.emoji?.id === activity.emoji?.id)
```

Comment:
```suggestion
        this.emoji?.id === activity.emoji?.id &&
        this.emoji?.name === activity.emoji?.name)
``` Also there it needs to check name

Reply:
Note for others: currently discussing in #8841.